### PR TITLE
Update DI parameters, aliases and services documentation

### PR DESCRIPTION
**Pull request by ChristophWurst**
The info was slightly outdated.

* Updated missing aliases
* Sort parameters alphabetically
* Fix namespaces of services
* Fix heading symbol length

### CODE REVIEW

File: developer_manual/basics/dependency_injection.rst [hunk 1]
```diff
@@ -310,45 +310,52 @@ The following parameter names and type hints can be used to inject core services
...
+* ``\OCP\Contacts\IManager``
+* ``\OCP\IDateTimeZone``
+* ``\OCP\IDBConnection``
+* ``\OCP\Diagnostics\IEventLogger``
+* ``\OCP\Diagnostics\IQueryLogger``
+* ``\OCP\Files\Config\IMountProviderCollection``
+* ``\OCP\Files\IRootFolder``
+* ``\OCP\IGroupManager``
+* ``\OCP\IL10N``
+* ``\OCP\ILogger``
```
Comment (juliushaertl) [inline developer_manual/basics/dependency_injection.rst:342]: Should we rather list the LoggerInterface here?

Comment (ChristophWurst) [reply developer_manual/basics/dependency_injection.rst:342]: Yes, makes sense. And also mark the ILogger as deprecated.

I did not check this list. There is a lot more that can be injected these days and I don't know how to get a reliable list for that info. This has to do with auto-wiring. In the past it was easier to see what is registered on ever app container.

Comment (juliushaertl) [reply developer_manual/basics/dependency_injection.rst:342]: Right, getting a full list is probably hard

**Pull request merged**